python-pytest.el

```diff
@@ -119,41 +119,50 @@ When non-nil only ‘test_foo()’ will match, and nothing else."
+  :class 'transient-option
+  :key "=t"
+  :argument "--tb="
+  :choices '("long" "short" "line" "native" "no"))
+
 ;;;###autoload (autoload 'python-pytest-popup "python-pytest" nil t)
-(magit-define-popup python-pytest-popup
+(define-transient-command python-pytest-popup ()
   "Show popup for running pytest."
-  'python-pytest
-  :switches
-  '((?c "color" "--color" t)
-    (?d "run doctests" "--doctest-modules")
-    (?f "failed first" "--failed-first")
-    (?l "show locals" "--showlocals")
-    (?p "debug on error" "--pdb")
-    (?q "quiet" "--quiet")
-    (?s "do not capture output" "--capture=no")
-    (?t "do not cut tracebacks" "--full-trace")
-    (?v "verbose" "--verbose")
-    (?w "very verbose" "-vv")
-    (?x "exit after first failure" "--exitfirst"))
-  :options
-  '((?k "only names matching expression" "-k")
-    (?m "only marks matching expression" "-m")
-    (?t "traceback style" "--tb=" python-pytest--choose-traceback-style)
-    (?x "exit after N failures or errors" "--maxfail="))
-  :actions
-  '("Run tests"
-    (?t "Test all" python-pytest)
-    (?r "Repeat last test run" python-pytest-repeat)
-    (?x "Test last-failed" python-pytest-last-failed)
-    "Run tests for specific files"
-    (?f "Test file (dwim)" python-pytest-file-dwim)
-    (?F "Test this file" python-pytest-file)
-    (?m "Test multiple files" python-pytest-files)
-    "Run tests for current function/class"
-    (?d "Test def/class (dwim)" python-pytest-function-dwim)
-    (?D "Test this def/class" python-pytest-function))
-  :max-action-columns 3
-  :default-action 'python-pytest-repeat)
+  :man-page "pytest"
+  :incompatible '(("--verbose" "--verbose --verbose"))
+  :value '("--color")
+  ["Switches"
+   ("-c" "color" "--color")
+   ("-d" "run doctests" "--doctest-modules")
+   ("-f" "failed first" "--failed-first")
+   ("-l" "show locals" "--showlocals")
+   ("-p" "debug on error" "--pdb")
+   ("-q" "quiet" "--quiet")
+   ("-s" "do not capture output" "--capture=no")
+   ("-t" "do not cut tracebacks" "--full-trace")
+   ("-v" "verbose" ("-v" "--verbose"))
+   ("-w" "very verbose" ("-vv" "--verbose --verbose"))
+   ("-x" "exit after first failure" "--exitfirst")]
+  ["Options"
+   ("=k" "only names matching expression" "-k=")
+   ("=m" "only marks matching expression" "-m=")
+   (python-pytest-popup:--tb)
+   ("=x" "exit after N failures or errors" "--maxfail=")]
+  [["Run tests"
+    ("t" "Test all" python-pytest)
+    ("r" "Repeat last test run" python-pytest-repeat)
+    ("x" "Test last-failed" python-pytest-last-failed)]
+   ["Run tests for specific files"
+    ("f" "Test file (dwifdsafjda;sfjsdla;kfjm)" python-pytest-file-dwim)
+    ("F" "Test this file" python-pytest-file)
+    ("m" "Test multiple files" python-pytest-files)]
+   ["Run tests for current function/class"
+    ("d" "Test def/class (dwim)" python-pytest-function-dwim)
+    ("D" "Test this def/class" python-pytest-function)]])
+
+(defun python-pytest-arguments ()
```

Comment:
is this function actually needed?

Reply:
I can remove it and replace all usages below. Again - this is how I noticet `magit` did the switch. I guess such a function was the convention that `magit-popup` has been using.

Reply:
yeah it was the original api to obtain the current state, with an associated customizable variable that would define defaults.

let's leave it in place to keep it simple. it's only if internally (or should be) anyway

Reply:
Leaving the function and will also move to helpers.